### ISSUE TITLE
Add higher level abstractions for Map and Sprite?

### ISSUE THREAD

**Issue by trosel** [DimChtz/rust-gamemap#1]
Any thought about making a `Map` be a list of `Option<Sprite>` instead of a list of coords? That way you can query if pieces are in certain locations for collision detection?

**Comment by DimChtz** [comment]
Yep, I like this. A new struct Sprite containing the location (x, y) and the display character, and a function called when 2 sprites collide.

**Comment by trosel** [comment]
@DimChtz What really might help flesh this out is an examples directory that shows how you intend it to be used.
